keel-web/src/main/kotlin/com/netflix/spinnaker/keel/apidocs/RegisteredExtensionModelConverter.kt

```diff
@@ -0,0 +1,32 @@
+package com.netflix.spinnaker.keel.apidocs
+
+import com.netflix.spinnaker.keel.api.support.ExtensionRegistry
+import io.swagger.v3.core.converter.AnnotatedType
+import io.swagger.v3.core.converter.ModelConverter
+import io.swagger.v3.core.converter.ModelConverterContext
+import io.swagger.v3.core.util.RefUtils.constructRef
+import io.swagger.v3.oas.models.media.Discriminator
+import io.swagger.v3.oas.models.media.Schema
+import org.springframework.stereotype.Component
+
+@Component
+class RegisteredExtensionModelConverter(
+  private val extensionRegistry: ExtensionRegistry
+) : BaseModelConverter() {
+  override fun resolve(
+    annotatedType: AnnotatedType,
+    context: ModelConverterContext,
+    chain: MutableIterator<ModelConverter>
+  ): Schema<*>? =
+    if (annotatedType.rawClass in extensionRegistry.baseTypes()) { val extensionTypes = extensionRegistry.extensionsOf(annotatedType.rawClass)
+      context.defineSchemaAsOneOf(annotatedType.rawClass, extensionTypes.values.toList())
+        .also { schema ->
+            schema.discriminator = Discriminator()
+              .propertyName("type") // TODO: is this a broken assumption?
```

Comment:
I have a faint recollection that we may have used a different property name somewhere... In the events class tree maybe? I can try and take a closer look.

Reply:
I checked all the ones that are actually registered using the `ExtensionRegistry` and the only outlier is `ResourceSpec` which is handled separately anyway and never hits this bit of code.